src/Lean/Elab/PreDefinition/Structural.lean

```diff
@@ -107,6 +109,7 @@ structure State where
      We use this information to generate the auxiliary `_sunfold` definition needed by the smart unfolding
      technique used at WHNF. -/
   matcherBelowDep : NameSet := {}
+  addMatchers : Array $ MetaM Unit := #[]
```

Comment:
Could you please add a comment explaining why you had to add `addMatchers`? I know why you did it, but I will probably forget.